editor/src/components/canvas/canvas-strategies/strategies/set-padding-strategy.spec.browser2.tsx

```diff
@@ -258,6 +259,155 @@ describe('Padding resize strategy', () => {
+
+    expect(getPrintedUiJsCode(editor.getEditorState())).toEqual(
+      makeTestProjectCodeWithLongHandPaddingValues({
+        paddingTop: `'10px'`,
+        paddingBottom: `'10px'`,
+        paddingLeft: `'10px'`,
+      }),
+    )
+  })
+
+  it('paddingBottom can be removed by dragging', async () => {
+    const dragDeltaToZero = -100
+    const editor = await renderTestEditorWithCode(
+      makeTestProjectCodeWithStringPaddingValues('10px 10px 10px 10px'),
+      'await-first-dom-report',
+    )
+
+    await testPaddingResizeForEdge(editor, dragDeltaToZero, 'bottom', 'precise')
+    await editor.getDispatchFollowUpActionsFinished()
+
+    await editor.getDispatchFollowUpActionsFinished()
+
+    expect(getPrintedUiJsCode(editor.getEditorState())).toEqual(
+      makeTestProjectCodeWithLongHandPaddingValues({
+        paddingTop: `'10px'`,
+        paddingLeft: `'10px'`,
+        paddingRight: `'10px'`,
+      }),
+    )
+  })
+
+  it('paddingTop can be readded by dragging', async () => {
+    const dragDeltaToZero = 10
+    const editor = await renderTestEditorWithCode(
+      makeTestProjectCodeWithLongHandPaddingValues({
+        paddingBottom: `'10px'`,
+        paddingLeft: `'10px'`,
+        paddingRight: `'10px'`,
+      }),
+      'await-first-dom-report',
+    )
+
+    await testPaddingResizeForEdge(editor, dragDeltaToZero, 'top', 'precise')
+    await editor.getDispatchFollowUpActionsFinished()
+
+    await editor.getDispatchFollowUpActionsFinished()
+
+    expect(getPrintedUiJsCode(editor.getEditorState())).toEqual(
+      makeTestProjectCodeWithStringPaddingValues('10px 10px 10px 10px'),
+    )
+  })
+
+  it('paddingTop can be removed by dragging when longhand props are present', async () => {
+    const dragDeltaToZero = -100
+    const editor = await renderTestEditorWithCode(
+      makeTestProjectCodeWithLongHandPaddingValues({
+        paddingTop: `'10px'`,
+        paddingLeft: `'10px'`,
+        paddingRight: `'10px'`,
+      }),
+      'await-first-dom-report',
+    )
+
+    await testPaddingResizeForEdge(editor, dragDeltaToZero, 'top', 'precise')
+    await editor.getDispatchFollowUpActionsFinished()
+
+    expect(getPrintedUiJsCode(editor.getEditorState())).toEqual(
+      makeTestProjectCodeWithLongHandPaddingValues({
+        paddingLeft: `'10px'`,
+        paddingRight: `'10px'`,
+      }),
+    )
+  })
+
+  it('paddingTop can be readded when only two longhands are present', async () => {
+    const dragDeltaToZero = 10
+    const editor = await renderTestEditorWithCode(
+      makeTestProjectCodeWithLongHandPaddingValues({
+        paddingLeft: `'10px'`,
+        paddingRight: `'10px'`,
+      }),
+      'await-first-dom-report',
+    )
+
+    await testPaddingResizeForEdge(editor, dragDeltaToZero, 'top', 'precise')
+    await editor.getDispatchFollowUpActionsFinished()
+
+    expect(getPrintedUiJsCode(editor.getEditorState())).toEqual(
+      makeTestProjectCodeWithLongHandPaddingValues({
+        paddingTop: `10`,
```

Comment:
It seems we have some inconsistent behaviour here. If we have 3 padding values set (without units) and then drag to add the 4th, we squash them into a single `padding` prop that explicitly provides `px` as the unit. Then remove two prop and the padding is split into the individual props with `px` still. Add one back in and that new one is unitless. Add the 4th again and they all have `px` again. It's a minor thing, but I think it's worth either always explicitly adding `px` or never adding it.

Reply:
it was easier to implement it this way but it is in fact better UX if `px` values are used consistently, I added it